### PR TITLE
Dont check for NaTType, just NaT

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/scalar/test_timedelta.py [hunk 1]
```diff
@@ -572,7 +572,7 @@ def test_implementation_limits(self):
         assert max_td.value == np.iinfo(np.int64).max
 
         # Beyond lower limit, a NAT before the Overflow
-        assert isinstance(min_td - Timedelta(1, 'ns'), NaTType)
+        assert min_td - Timedelta(1, 'ns') is NaT
```
Comment (jreback) [inline pandas/tests/scalar/test_timedelta.py]: use parens

**Review by jreback — CHANGES_REQUESTED** [review]
small change, other lgtm.

File: pandas/io/packers.py [hunk 2]
```diff
@@ -470,7 +469,7 @@ def encode(obj):
                     }
 
     elif isinstance(obj, (datetime, date, np.datetime64, timedelta,
-                          np.timedelta64, NaTType)):
```
Comment (jreback) [inline pandas/io/packers.py]: FYI: 

```
In [40]: np.timedelta64('nat') is pd.NaT
Out[40]: False

In [41]: pd.Timedelta(np.timedelta64('nat')) is pd.NaT
Out[41]: True
```

this is reason for the complex type checking around NaTs; numpy has a 'nat' like value embedded in np.timedelta64/np.datetime64 (however its not very useful).

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. merge on green.

**Comment by jbrockmendel** [comment]
Test error is in `pyarrow`, which I haven't looked at closely.  Does it instantiate a new instance of `NaTType` for some reason?

**Comment by jreback** [comment]
no just rebase

**Comment by jreback** [comment]
thanks!

**Pull request merged**